### ISSUE TITLE
Provide default data for Carousel (store-driven)

### ISSUE THREAD

**Issue by jhpedemonte** [maqetta/maqetta#515]
Need to provide some sort of default, sample data for the store-driven dojox.mobile widgets.  These include:
- ~~RoundRectDataList~~ #461
- ~~EdgeToEdgeDataList~~ #461
- Carousel

Perhaps just use sample JSON files (and related content) from test suite, and drop an `ItemFileReadStore` for these widgets, which references JSON file.

@billreed63

**Comment by jhpedemonte** [comment]
Fixing this for `ComboBox` may fix #525.

**Comment by jhpedemonte** [comment]
@billreed63, looks like the 3 widgets mentioned in the summary can all use **ItemFileReadStore**.  Looking in widgets.json, I assume I'd have to create a `"data"` prop and a custom **CreateTool** object in order to make this work, correct?  Is there an existing widget which is the simplest, which I can use as a guideline?

**Comment by billreed63** [comment]
DataGrid is a good example

**Comment by jhpedemonte** [comment]
The two *List widgets were covered by #461.  Using this bug to track Carousel.

---

For now, I'm going to move Carousel to untested, until we can figure out its data situation.

I think the best solution would be something like what Bill did for RoundRectDataList, where there is some **default data** but we also provide a **sample JSON** (and associated resources) which the user can use with Carousel.

The problem here is providing the default data.  The data store for Carousel specifies text as well as image URLs:

``` js
{
    items: [
        {src:"images/dish1.jpg", value:"dish", headerText:"dish"},
        {src:"images/glass1.jpg", value:"glass", headerText:"glass"},
        {src:"images/stone1.jpg", value:"stone", headerText:"stone"},
        {src:"images/shell1.jpg", value:"shell", headerText:"shell"}
    ]
}
```

Which means our default data consists of _text_ and _images_.  Currently, we don't support having the library bundle supply anything other than metadata and source files.

One solution would be to add support for OAM's `<require type="image">`.  Something like:

``` js
    "require": [
        ...
        {
            "type": "image",
            "src": "../../resources/carousel/dish1.jpg",
            "target": "images/"
        }
    ],
```

This would copy **dish1.jpg** into a directory called **images**, which would be a sibling to the file the user is working on.

**Comment by jhpedemonte** [comment]
I forgot that for IconContainer, I just used a base64 image.  I can do the same here.

I also made use of EdgeToEdgeDataListCreateTool, since it is generic enough and creates the required data store*.

With this, a Carousel with 3 images is created.  Changing props through Property editor fails, though. @billreed63, what am I missing here?  If we can get that part working, can probably move Carousel out of **untested**.

\* _This should all be refactored as much as possible in the future, so we don't always have to create an extra object in order to create the data store.  Perhaps with the "sibling widget" issue #731._

**Comment by billreed63** [comment]
Because the widget uses a datastore some special magic has to happen with the widget.store whenever the widget is modified. So you need a helper also "helper": "davinci.libraries.dojo.dojox.mobile.EdgeToEdgeDataListHelper", might be a good model for this.

**Comment by jhpedemonte** [comment]
Moving to **M3** to finish remainder of work.

**Comment by jhpedemonte** [comment]
-> M6

**Comment by doronrosenberg** [comment]
some fixes for this were checked in 575d3f8d3e39f184ad1004331fe648a44a066c32 by mistake as well.
